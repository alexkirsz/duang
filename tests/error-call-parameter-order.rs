// this test check error message when unnamed parameter comes after named ones.
use duang::duang;

duang!(
pub fn foo<T>(a: T,b: f64 = 13.0, c: T = a*a) -> (T,f64,T)
where
  T: std::ops::Mul<T, Output = T>,
  T: std::fmt::Display,
  T: Copy,
{
  (a,b,c)
}
);

fn main() {
  foo!(a=1,0.1);
}
