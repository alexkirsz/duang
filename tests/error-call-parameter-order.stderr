error: Unnamed parameter must be ahead of named parameter.
  --> $DIR/error-call-parameter-order.rs:16:12
   |
16 |   foo!(a=1,0.1);
   |   ---------^^^-- in this macro invocation
